lint/rule_name.go

```diff
@@ -29,12 +30,41 @@ const nameSeparator string = "::"
 
 var ruleNameValidator = regexp.MustCompile("^([a-z0-9][a-z0-9-]*(::[a-z0-9][a-z0-9-]*)?)+$")
 
-// NewRuleName creates a RuleName from segments.
+// NewRuleName creates a RuleName from an AIP number and a unique name within
+// that AIP.
+func NewRuleName(aip int, name string) RuleName {
+	// Determine the group.
+	group := ""
+	if aip > 0 && aip < 1000 {
+		group = "core"
+	}
+
+	// Sanity check: If the group does not exist, complain.
+	if group == "" {
+		panic("Invalid AIP; no available group.")
+	}
+
+	// Get the AIP as a four-character string.
+	aipStr := fmt.Sprintf("%d", aip)
+	for len(aipStr) < 4 {
+		aipStr = "0" + aipStr
+	}
+
+	// Return the rule name.
+	return NewGenericRuleName(group, aipStr, name)
+}
+
+// NewGenericRuleName creates a RuleName from segments.
 // It will join the segments with the "::" separator.
-func NewRuleName(segments ...string) RuleName {
+//
+// Use this function for rules not covered in public AIPs.
+func NewGenericRuleName(segments ...string) RuleName {
```

Comment:
I assumed that this will only be used in the tests. Can we add a "test" group? Otherwise, this function is public, there is no way to prevent misusing it other than code review.

Reply:
Actually there is a valid use case for this -- yours. :-)

When someone writes "private" rules (like what your team will do), their private AIPs will not be in the public index, and `NewGenericRuleName` will be needed.

Reply:
I think this function can be created anywhere, like
```
package myrules
func NewMyRuleName(segments ...string) lint.RuleName { ... }
```
Indeed, I guess people can even do
```
testRuleName := lint.RuleName("test::test-rule-name")

```
So we don't need to provide it now, and we can wait until people yell at us for it.

Reply:
Fair enough.

Reply:
Done.